ecosystem/sep-direct-payment.md

```diff
@@ -0,0 +1,394 @@
+`refunded` | boolean | (optional) Should be true if the transaction was refunded. Not including this field means the transaction was not refunded.
+`required_info_message` | string | (optional) A human readable message indicating any errors that require updated information from the sender
+`required_info_updates` | object | (optional) A set of fields that require upate from the sender, in the same format as described in [/info](#info).  Fields should be broken out by sender, receiver, and transacton as in /info.
+
+`status` should be one of:
+
+* `pending_sender` -- awaiting payment to be initiated by sending anchor
+* `pending_stellar` -- transaction has been submitted to Stellar network, but is not yet confirmed.
+* `pending_info_update` -- certain pieces of information need to be updated by the sending anchor.  See [pending info update](#pending-info-update) section
+* `pending_receiver` -- payment is being processed by the receiving anchor
+* `pending_external` -- payment has been submitted to external network, but is not yet confirmed.
+* `completed` -- deposit/withdrawal fully completed.
+* `error` -- catch-all for any error not enumerated above.
+
+Example response:
+
+```json
+{
+  "transaction": {
+      "id": "82fhs729f63dh0v4",
+      "status": "pending_external",
+      "status_eta": 3600,
+      "external_transaction_id": "ABCDEFG1234567890",
+      "amount_in": "18.34",
+      "amount_out": "18.24",
+      "amount_fee": "0.1",
+      "started_at": "2017-03-20T17:05:32Z"
+    }
+}
+```
+
+
+```json
+{
+  "transaction": {
+      "id": "82fhs729f63dh0v4",
+      "status": "pending_info_update",
+      "status_eta": 3600,
+      "external_transaction_id": "ABCDEFG1234567890",
+      "amount_in": "18.34",
+      "amount_out": "18.24",
+      "amount_fee": "0.1",
+      "started_at": "2017-03-20T17:05:32Z",
+      "required_info_message": "The bank reported an incorrect name for the receiver, please ensure the name matches legal documents",
+      "required_info_fields": {
+         "receiver": {
+            "first_name":"The receiver's first name",
+            "last_name":"The receiver's last name"
+         }
+      }
+    }
+}
+```
+
+If the transaction cannot be found, the endpoint should return a `404 NOT FOUND` result.
+
+#### Pending info update
+
+In certain cases the receiver might need to request updated information, for example if the bank tells them that the provided receiver name is incorrect, or missing a middle initial.  In this case, the transaction should go into the `pending_info_update` state until the sender provides updates.
+
+### Update
+
+The `/update` endpoint allows for updating certain pieces of information that need to be corrected.  For example a bank may reject a deposit because a name is mis-spelled, or a middle initial is missing.  This allows transactions to be updated instead of errored and refunded.
```

Comment:
I think a second paragraph outlining *when* the sender should be able to use this endpoint would be helpful. Should anchors implement the endpoint so that it will always fail if not in `pending_info_update` state or should the sender be allowed to update eagerly?

(Not sure how realistic that scenario is in the real world, but at least technically this edge case seems feasible)

Reply:
Added so that if the sender tries to update when the receiver doesn't allow it it should error